services/analytics/api/src/main/kotlin/io/element/android/services/analytics/api/extensions/JoinedRoomExt.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.services.analytics.api.extensions
+
+import im.vector.app.features.analytics.plan.JoinedRoom
+import io.element.android.libraries.core.bool.orFalse
+import io.element.android.libraries.matrix.api.core.MatrixPatterns
+import io.element.android.libraries.matrix.api.room.MatrixRoom
+
+fun Long?.toAnalyticsRoomSize(): JoinedRoom.RoomSize {
+    return when (this) {
+        null,
+        2L -> JoinedRoom.RoomSize.Two
+        in 3..10 -> JoinedRoom.RoomSize.ThreeToTen
+        in 11..100 -> JoinedRoom.RoomSize.ElevenToOneHundred
+        in 101..1000 -> JoinedRoom.RoomSize.OneHundredAndOneToAThousand
+        else -> JoinedRoom.RoomSize.MoreThanAThousand
+    }
+}
+
+fun MatrixRoom?.toAnalyticsJoinedRoom(trigger: JoinedRoom.Trigger?): JoinedRoom {
+    return JoinedRoom(
+            isDM = this?.isDirect.orFalse(),
+            isSpace = MatrixPatterns.isSpaceId(this?.roomId?.value),
+            roomSize = this?.joinedMemberCount?.toAnalyticsRoomSize() ?: JoinedRoom.RoomSize.Two,
+            trigger = trigger
+    )
+}
```

Comment:
The nullability and defaults here seem a bit weird to me. If the `MatrixRoom` is `null` we'll log that it's not a DM and has two people in it... is that not just completely bad data?

This could just have a non-null receiver (i.e. `MatrixRoom` not `MatrixRoom?`) which would probably fix that?

Reply:
You right. I updated that.